### PR TITLE
[llvm-cov] - Output better error message when the error kind is `coveragemap_error::malforme`.

### PR DESCRIPTION
**Pull request by MaggieYingYi**
The current llvm-cov error message for kind `coveragemap_error::malforme`,  just gives the issue kind without any reason for what caused the issue. This patch is aimed at improving the llvm-cov error message to help identify what caused the issue.

### CODE REVIEW

File: llvm/lib/ProfileData/Coverage/CoverageMappingReader.cpp [hunk 1]
```diff
@@ -634,7 +648,9 @@ class VersionedCovMapFuncRecordReader : public CovMapFuncRecordReader {
 
     // Get the filenames.
     if (CovBuf + FilenamesSize > CovBufEnd)
-      return make_error<CoverageMapError>(coveragemap_error::malformed);
+      return make_error<CoverageMapError>(
+          coveragemap_error::malformed,
+          "filenames section is greater than buffer size");
```
Comment (MaskRay) [inline llvm/lib/ProfileData/Coverage/CoverageMappingReader.cpp:653]: larger

File: llvm/lib/ProfileData/Coverage/CoverageMapping.cpp [hunk 2]
```diff
@@ -925,26 +926,45 @@ LineCoverageIterator &LineCoverageIterator::operator++() {
...
-static std::string getCoverageMapErrString(coveragemap_error Err) {
+static std::string getCoverageMapErrString(coveragemap_error Err,
+                                           const std::string &ErrMsg = "") {
+  std::string Msg;
+  raw_string_ostream OS(Msg);
+
   switch (Err) {
   case coveragemap_error::success:
-    return "Success";
+    OS << "Success";
```
Comment (MaskRay) [inline llvm/lib/ProfileData/Coverage/CoverageMapping.cpp:936]: It seems that most message are untested?

Note, the diagnostic formatting follows Clang. We don't use capitalization: https://llvm.org/docs/CodingStandards.html#error-and-warning-messages

**Review by MaskRay — APPROVED** [review]

File: llvm/lib/ProfileData/Coverage/CoverageMappingReader.cpp [hunk 3]
```diff
@@ -613,7 +625,9 @@ class VersionedCovMapFuncRecordReader : public CovMapFuncRecordReader {
     using namespace support;
 
     if (CovBuf + sizeof(CovMapHeader) > CovBufEnd)
-      return make_error<CoverageMapError>(coveragemap_error::malformed);
+      return make_error<CoverageMapError>(
+          coveragemap_error::malformed,
+          "coverage mapping header section is greater than buffer size");
```
[the remark this answers is not in the document]
Comment (MaggieYingYi) [reply llvm/lib/ProfileData/Coverage/CoverageMappingReader.cpp]: Fixed, thanks.

File: llvm/lib/ProfileData/Coverage/CoverageMappingReader.cpp (hunk 1 above)
Comment (MaggieYingYi) [reply llvm/lib/ProfileData/Coverage/CoverageMappingReader.cpp]: Fixed, thanks.

File: llvm/lib/ProfileData/Coverage/CoverageMapping.cpp (hunk 2 above)
Comment (MaggieYingYi) [reply llvm/lib/ProfileData/Coverage/CoverageMapping.cpp]: The diagnostic formatting of llvm-cov has been updated to follow Clang in the commit https://github.com/llvm/llvm-project/pull/65264/commits/521abebc6c3a64f7c4060564300f9725a5a1a751.

**Comment by abrachet** [comment]
I'm seeing failures in Linux/binary-id-lookup.c after this patch. Looks like the messages need to be updated in that test as well. See https://ci.chromium.org/ui/p/fuchsia/builders/toolchain.ci/clang-linux-x64/b8770695790398874177/overview

Would you mind taking a look?

**Comment by MaggieYingYi** [comment]
Thanks @abrachet, I am fixing it.

**Comment by MaggieYingYi** [comment]
Fixed in the commit  61d615404.

**Pull request merged**